apps/src/music/views/PanelContainer.jsx

```diff
@@ -0,0 +1,58 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {useSelector} from 'react-redux';
+import classNames from 'classnames';
+import moduleStyles from './panelContainer.module.scss';
+
+/**
+ * A container for a top-level panel with a header.  The content of the panel
+ * is provided as children.  Whether the header is shown is determined by an
+ * external state value.  The panel container always occupies the full size of
+ * its parent; this means that the main scene is responsible for allocating
+ * the layout of all the panels.
+ */
+const PanelContainer = ({id, headerText, children}) => {
+  const isHeadersShowing = useSelector(state => state.music.isHeadersShowing);
+
+  return (
+    <div
+      className={classNames('panelContainer', moduleStyles.panelContainer)}
+      id={id}
+    >
+      {isHeadersShowing && (
```

Comment:
nit: could invert this to just return {children} if isHeadersShowing if false?

Reply:
Could you elaborate?  We always want to return the parent `div`, and then we always want to return the `children`, but the optional part between them is the header.

Reply:
Ah okay, missed that children are always wrapped in the parent div. Not a huge deal then, but we could still do something like
```
if (!isHeadersShowing) {
   return (<div ...>{children}</div>);
}

return (
... // components with headers
);
```